### DISCUSSION TITLE
WEBGUI

### DISCUSSION THREAD

**Discussion by shanekuz** [wayne-love/sn_esp32#4] [Ideas]
I know the intention here is for MQTT to home automation but im thinking a web interface could be a good addition also.

**Comment by wayne-love** [comment]
@shanekuz 
> a web interface could be a good addition also

This would be a trivial feature to add, all the architecture is there to support this but at the moment I'm focusing on getting the spa controls exposed on MQTT.  If someone were to supply an HTML file with the appropriate javascript refresh built-in then I would be able to drop this into the code.... :-)

**Reply by shanekuz** [reply]
Happy to help, im not as good a coder as you but if i create a separate file and you integrate showing how to set and read i will work on making it look OK and adding as you add.

**Reply by wayne-love** [reply]
Sorry, that made me laugh!  I'm not a coder I'm a hack!  I'd love someone with real c++ experience to have a go at reviewing my code and providing constructive critism.

I'm currently time-poor, my current work is around getting the jets working.  I'll drop a placeholder HTML page in for displaying the status of the spa after this.  Updating values will come after that.

**Reply by shanekuz** [reply]
Ha your defiantly better than me I can understand it and string other peoples examples togeather to get what I want and edit.

**Reply by shanekuz** [reply]
I have a couple of esp32s on the way will make up a circuit board design too so people can just order on line with parts from Ali. When you have time drop the html starter page in and i will go from there.

**Reply by wayne-love** [reply]
Still working on pumps, having a dynamic number of pumps caused a number of headaches.  Getting much closer, should have the pumps sorted by the weekend and then onto the HTML.

**Reply by wayne-love** [reply]
@shanekuz - Check out the latest commit.  Have added auto refreshing web status page.  The details should be pretty easy to pick out from the code, have a look at WebUI.h and indexPageTemplate.  For the syntax on-field substitutions google sprintf.  Actual substitutions are defined in the WebUI::begin() function.

**Reply by shanekuz** [reply]
Thanks will take a look and work on it.

**Reply by Mjprinsloo1919** [reply]
Hi All. Just wondering if the WebUI was ever hooked up?

**Reply by wayne-love** [reply]
There is a rudimentary UI in place, if someone wants to flesh it out then be my guest but it is not a priority for me.
